source/state_representation/include/state_representation/Robot/JointState.hpp

```diff
@@ -260,13 +415,17 @@ inline void JointState::set_state_variable(Eigen::VectorXd& state_variable, cons
   this->set_state_variable(state_variable, Eigen::VectorXd::Map(new_value.data(), new_value.size()));
 }
 
+inline void JointState::set_all_state_variables(const Eigen::VectorXd& new_values) {
+  this->set_positions(new_values.segment(0, this->get_size()));
+  this->set_velocities(new_values.segment(this->get_size(), this->get_size()));
+  this->set_accelerations(new_values.segment(2 * this->get_size(), this->get_size()));
+  this->set_torques(new_values.segment(3 * this->get_size(), this->get_size()));
+}
+
 inline JointState& JointState::operator=(const JointState& state) {
   State::operator=(state);
   this->set_names(state.get_names());
-  this->set_positions(state.get_positions());
-  this->set_velocities(state.get_velocities());
-  this->set_accelerations(state.get_accelerations());
-  this->set_torques(state.get_torques());
+  this->set_all_state_variables(state.get_all_state_variables());
   return (*this);
 }
 
```

Comment:
I can't make a comment down on line 458, but set_names should not just resize the vector in my opinion